### PR TITLE
itstool: fix double-shebang issue on macOS

### PR DESCRIPTION
**Pull request by burke**
I haven't tried this on Linux, where it should have no effect, but it does make make libgda (which is [failing on darwin](https://hydra.nixos.org/build/107729826)) work correctly on my machine.

### CODE REVIEW

**Comment by FRidh** [comment]
We need a generic solution for this and should not be fixing individual packages. cc @NixOS/darwin-maintainers any ideas? Maybe a darwin-specific hook that patches all shebangs to prefix `usr/bin/env`?

**Comment by lavoiesl** [comment]
Would fix https://github.com/NixOS/nixpkgs/issues/73762

File: pkgs/development/tools/misc/itstool/default.nix [hunk 1]
```diff
@@ -10,6 +10,22 @@ stdenv.mkDerivation rec {
...
+  # withPackages' shebang is "#!#{bash}/bin/bash -e
+  #
+  # macOS won't allow the target of a shebang to be an interpreted script,
+  # causing bin/itstool to get interpreted as bash.
+  #
+  # By prefixing /usr/bin/env to the shebang, we have env fork/exec the python
+  # wrapper, which is perfectly happy to execute an interpreted script.
+  #
+  # However, we don't want to do this on Linux, which only allows one argument
+  # in a shebang.
```
Comment (matthewbauer) [inline pkgs/development/tools/misc/itstool/default.nix]: This sounds like something we can update in patch-shebangs.sh. I didn't realize that /usr/bin/env worked differently! For instance, skipping it here: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/patch-shebangs.sh#L64

**Comment by vcunat** [comment]
I _think_ this is what is blocking the nixpkgs-20.03-darwin channel: [`darwin-tested`](https://hydra.nixos.org/job/nixpkgs/nixpkgs-20.03-darwin/darwin-tested#tabs-constituents) -> `gimp` -> `shared-mime-info`.  /cc @NixOS/nixos-release-managers.

**Comment by jtojnar** [comment]
Would just patching `newInterpreterLine` on Darwin to `@coreutils-or-wherever-env-is-on-mac@/bin/env -S $newPath $args` work? `-S` should split the arguments according to man page in [this answer](https://apple.stackexchange.com/questions/297533/does-macos-have-usr-bin-env) so it should work even with multiple arguments. cc @NixOS/darwin-maintainers

**Review by Mic92 — CHANGES_REQUESTED** [review]
There is a better fix for this using `wrapPython`:

Please cherry-pick this commit instead: fa247de4d70015608f8f70a8192757da6595488f

**Comment by LnL7** [comment]
@Mic92 That looks like the way to go to me.

While I guess changing patchShebangs would work around this I'd much rather avoid double wrapping in general, it's rather ugly IMHO.

**Comment by burke** [comment]
Cherry-pick completed. Thanks!

**Comment by domenkozar** [comment]
@GrahamcOfBorg build itstool

**Comment by domenkozar** [comment]
Should this go to staging?

**Comment by worldofpeace** [comment]
> Should this go to staging?

Yes, it appears to rebuild the world (not just on darwin). I think we can stand a mass rebuild just on darwin though.

**Pull request merged**